inventory_management_system_api/repositories/manufacturer.py

```diff
@@ -133,16 +133,20 @@ def delete(self, manufacturer_id: str, session: ClientSession = None) -> None:
         if result.deleted_count == 0:
             raise MissingRecordError(f"No manufacturer found with ID: {str(manufacturer_id)}")
 
-    def _is_duplicate_manufacturer(self, code: str, session: ClientSession = None) -> bool:
+    def _is_duplicate_manufacturer(
+        self, code: str, manufacturer_id: CustomObjectId = None, session: ClientSession = None
+    ) -> bool:
         """
         Check if manufacturer with the same name already exists in the manufacturer collection
 
         :param code: The code of the manufacturer to check for duplicates.
+        :param manufacturer_id: The ID of the manufacturer to check if the duplicate manufacturer found is itself.
         :param session: PyMongo ClientSession to use for database operations
         :return `True` if duplicate manufacturer, `False` otherwise
         """
         logger.info("Checking if manufacturer with code '%s' already exists", code)
-        return self._manufacturers_collection.find_one({"code": code}, session=session) is not None
+        manufacturer_with_code = self._manufacturers_collection.find_one({"code": code}, session=session)
```

Comment:
Calling this `manufacturer` should be just fine I think
```suggestion
        manufacturer = self._manufacturers_collection.find_one({"code": code}, session=session)
```